python/lsst/resources/_resourceHandles/_httpResourceHandle.py

```diff
@@ -203,17 +204,13 @@ def read(self, size: int = -1) -> bytes:
         # in the file and also the current position we have got to in the
         # server.
         if "Content-Range" in resp.headers:
-            content_range = resp.headers["Content-Range"]
-            units, range_string = content_range.split(" ")
-            if units == "bytes":
-                range, total = range_string.split("/")
-                if "-" in range:
-                    _, end = range.split("-")
-                    end_pos = int(end)
-                    if total != "*" and end_pos >= int(total) - 1:
-                        self._eof = True
-            else:
-                self._log.warning("Requested byte range from server but instead got: %s", content_range)
```

Comment:
There's a slight behavior change here.  Previously if the units were anything but `bytes` it would only log a warning.  Now it throws an exception.  This should never happen in real life, and if it does the returned data is almost certainly wrong so we want to bail anyway.